Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py

```diff
@@ -44,205 +19,200 @@ def default(self, obj):
+    aws_role_arn = params.get('roleArn')
+    aws_role_session_name = params.get('roleSessionName')
+    aws_default_region = params.get('defaultRegion')
+    aws_role_session_duration = params.get('sessionDuration')
+    aws_access_key_id = demisto.get(params, 'credentials.identifier') or params.get('access_key')
+    aws_secret_access_key = demisto.get(params, 'credentials.password') or params.get('secret_key')
+    verify_certificate = not params.get('insecure', True)
+    timeout = params.get('timeout')
+    retries = params.get('retries', 5)
 
-    elif demisto.command() == 'aws-athena-stop-query':
-        stop_query_command(demisto.args())
+    try:
+        demisto.debug(f"Command being called is '{command}'.")
+
+        aws_client = AWSClient(aws_default_region=aws_default_region, aws_role_arn=aws_role_arn,
+                               aws_role_session_name=aws_role_session_name, aws_role_session_duration=aws_role_session_duration,
+                               aws_role_policy=None, aws_access_key_id=aws_access_key_id,
+                               aws_secret_access_key=aws_secret_access_key,
+                               verify_certificate=verify_certificate, timeout=timeout, retries=retries)
+
+        client = aws_client.aws_session(
+            service=AWS_SERVICE_NAME,
+            region=args.get('region'),
+            role_arn=args.get('roleArn'),
+            role_session_name=args.get('roleSessionName'),
+            role_session_duration=args.get('roleSessionDuration'),
+        )
+
+        result: str | CommandResults
+
+        if demisto.command() == 'test-module':
+            response = client.list_named_queries()
+            if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+                result = 'ok'
+
+            else:
+                result = CommandResults(readable_output=f'Error: {response}')
+
+        elif demisto.command() == 'aws-athena-start-query':
+            result = start_query_execution_command(args=args, client=client)
+
+        elif demisto.command() == 'aws-athena-stop-query':
+            result = stop_query_command(args=args, client=client)
+
+        elif demisto.command() == 'aws-athena-get-query-execution':
+            result = get_query_execution_command(args=args, client=client)
+
+        elif demisto.command() == 'aws-athena-get-query-results':
+            result = get_query_results_command(args=args, client=client)
+
+        else:
+            raise NotImplementedError(f'Command "{command}" is not implemented.')
 
-    elif demisto.command() == 'aws-athena-get-query-execution':
-        get_query_execution_command(demisto.args())
+        return_results(result)
 
-    elif demisto.command() == 'aws-athena-get-query-results':
-        get_query_results_command(demisto.args())
+    except Exception as e:
+        return_error(f'Error: {e}')
+        demisto.error(f'Error: {e}\n\nTraceback:\n{traceback.format_exc()}')
```

Comment:
You don't need the traceback anymore, in any case it's unreachable, please remove.